### PR TITLE
add auto npm publishing in GH action

### PR DESCRIPTION
**Pull request by isaaccormack**
AHHHHHH FINALLY

This adds a step in the GH action on push to main to bump the minor npm version and publish the package

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@vikelabs/uvic-course-scraper",
-  "version": "0.1.0",
+  "version": "0.1.14",
```
Comment (isaaccormack) [inline package.json:3]: lol

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,83 +1,119 @@
...
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          name: codecov-umbrella # optional
+          fail_ci_if_error: true # optional (default = false)
+          verbose: true # optional (default = false)
+
+  publish-npm:
+    environment: deployment
+    runs-on: ubuntu-latest
```
Comment (aomi) [inline .github/workflows/main.yml:97]: if you don't add a `needs` it'll run before everything else.
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,83 +1,119 @@
...
+        uses: 'phips28/gh-action-bump-version@master'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Create .npmrc for auth token publishing
+        run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
+        env:
+          # token must be generated by admin account to vikelabs org, currently gnomi
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
```
Comment (aomi) [inline .github/workflows/main.yml:117]: copy the caching step from the other workflow.

File: .github/workflows/main.yml (hunk 2 above)
Comment (isaaccormack) [reply .github/workflows/main.yml:97]: ahh right, thanks

File: .github/workflows/main.yml (hunk 3 above)
Comment (isaaccormack) [reply .github/workflows/main.yml:117]: right

**Review by aomi — APPROVED** [review]

**Pull request merged**